templates/todo/api/csharp-sql-func/ListsFunctions.cs

```diff
@@ -0,0 +1,199 @@
+using System.Net;
+using Microsoft.Azure.Functions.Worker;
+using Microsoft.Azure.Functions.Worker.Http;
+using Microsoft.Extensions.Logging;
+using System.Text.Json;
+
+namespace SimpleTodo.Api;
+public class ListsFunctions
```

Comment:
We generally serve a Swagger UI at the root endpoint. For example, the [code](https://github.com/Azure/azure-dev/blob/db03632996c3185372720c3f38f885cf993f1aef/templates/todo/api/csharp-sql/Program.cs#L34) in csharp-sql.

Is it possible to attach a route to `root` for functions? Not a critical functionality, but I would love a consistent experience across all our templates.

Reply:
I'm looking into this, so far it seems like it isn't as straightforward as in the csharp-sql project. 

- Routing used to be possible with https://github.com/Azure/azure-functions-openapi-extension/blob/main/docs/shortening-swagger-ui-page-url.md but proxies are no longer recommended in functions v4 and APIM is.
- Serving openapi.yaml to swagger UI doesn't seem to be supported either and would have to use attribute classes like this https://github.com/Azure/azure-functions-openapi-extension/blob/main/samples/Microsoft.Azure.Functions.Worker.Extensions.OpenApi.FunctionApp.OutOfProc/StoreHttpTrigger.cs

cc @jongio

Reply:
@justinyoo

we have openapi.yaml in the root of our function dotnet-isolated project.  In other projects we use 





// Swagger UI

app.UseSwaggerUI(options => {

    options.SwaggerEndpoint("./openapi.yaml", "v1");

    options.RoutePrefix = "";

});



Do you know what the equivalent would be for function dotnet-isolated?





Could you have a look at the code here and let us know what we should do to get the root request "/" to render the openapi.yaml file in the swagger ui?

Reply:
@jongio You can modify `host.json` file of dotnet-isolated function app.

```jsonc
// host.json
{
  "extensions": {
    "http": {
      // Change this 'routePrefix' value from "api" (default) to ""
      "routePrefix": ""
    }
  }
}
```

Then your Swagger UI endpoint will be `http://localhost:7071/swagger/ui`.

Reply:
Oh, you want to call like `http://localhost:7071`, and it shows Swagger UI or swagger.json document? It might need a hacky workaround. Let me find out and get back to you.

Reply:
@jongio @madebygps I created a sample repo for you to take a look: https://github.com/justinyoo/azfunc-openapi-on-root

It needs a bit of work for your function app because, the function app runtime occupies the root page for other purpose. And there's no URL rewrite feature until the new version of Azure Functions v4 runtime that puts the proxy feature back in.

TL;DR &ndash;

1. Update `host.json` to remove the `api` prefix.
2. Update `local.settings.json` to disable the homepage.
3. Add another endpoint to redirect traffic from `/` to `/swagger/ui`.

Again, please check out the sample repository above as a reference.

Reply:
Great  Thanks for doing that!  @madebygps - If you don't get to it first, then I'll have a look tomorrow.

Reply:
Oh... the OpenAPI extension doesn't support that feature and will not do it. - loading an OpenAPI document and rendering it with Swagger UI. If you want to do that, you should implement at least two endpoints by yourself, without having to rely on the AzFunc OpenAPI extension.

Reply:
Sounds good to me.